google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,502 @@
+            Required. The gcs bucket.
+
+            Format: Given `gs://{bucket_name}/{optional_folder}/{operation_id}/` where `gcs_bucket_name={bucket_name}`.
+        gcs_prefix (str):
+            Required. The prefix to the location of the target folder.
+
+            Format: Given `gs://{bucket_name}/{optional_folder}/{target_folder}` where `gcs_prefix={optional_folder}/{target_folder}`.
+    Returns:
+        Tuple[list, list]:
+            base document.proto, revision shards.
+    """
+    base_shards = []
+    revision_shards = []
+
+    text, base_bytes, revision_bytes = _get_base_and_revision_bytes(
+        gcs_bucket_name, gcs_prefix
+    )
+    page_pb = documentai.Document.Page.pb()
+    pb = documentai.Document.pb()
+    for byte in base_bytes:
+        base_shards.append(
+            documentai.Document(pages=[page_pb.FromString(byte)], text=text)
+        )
+
+    for byte in revision_bytes:
+        revision_shards.append(pb.FromString(byte))
+
+    return base_shards, revision_shards
+
+
+def _get_revised_entites(
+    entities: List[documentai.Document.Entity],
+) -> Tuple[list, list]:
+    r"""Modifies the entities using the providence field in the entities.
+
+    Args:
+        entities (List[Entity]):
+            A list of documentai.Document.Entity.
+
+    Returns:
+        Tuple[list,list]:
+            modified entities, Entity history.
+    """
+    entities_array = []
+    history = []
+    for e in entities:
+        if (
+            e.provenance.type_ == _OP_TYPE.ADD
+            or e.provenance.type_ == _OP_TYPE.OPERATION_TYPE_UNSPECIFIED
+        ):
+            entities_array.append(Entity(documentai_entity=e))
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                }
+            )
+        elif e.provenance.type_ == _OP_TYPE.REMOVE:
+            entity = entities_array.pop(int(e.id) - 1)
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": entity.type_,
+                    "original_text": entity.mention_text,
+                }
+            )
+            del entity
+        elif e.provenance.type_ == _OP_TYPE.REPLACE:
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                    "replace_type": e.type_,
+                    "replace_text": e.mention_text,
```

Comment:
Here it looks like "replace_text" and "original_text" are always the same?

Reply:
Fixed